alpha/alpha-fsm-channel-kafka/src/main/java/org/apache/servicecomb/pack/alpha/fsm/channel/kafka/KafkaMessagePublisher.java

```diff
@@ -43,7 +44,12 @@ public void publish(Object data) {
         }
 
         try {
-            kafkaTemplate.send(topic, data).get();
+            if(data instanceof BaseEvent) {
+                BaseEvent event = (BaseEvent) data;
+                kafkaTemplate.send(topic, event.getGlobalTxId(), event).get();
+            }else{
+                kafkaTemplate.send(topic, data).get();
```

Comment:
If the message format does not meet expectations, we only need to print the error log and discard the message.

Reply:
This is used internally and should not be called externally, just because the parameters of the publish method defined in MessagePublisher define Object, so you need to convert the input object when you need to get the globalTxId.

Reply:
> This is used internally and should not be called externally, just because the parameters of the publish method defined in MessagePublisher define Object, so you need to convert the input object when you need to get the globalTxId.

So I understand that kafkaTemplate.send(topic, data).get() will never be executed

Reply:
fix check msg type